core/includes/config.inc

```diff
@@ -1232,6 +1241,337 @@ interface ConfigStorageInterface {
+          ->condition('name', $name)
+          ->condition('ctime', 0)
+          ->execute();
+      }
+    }
+    catch (\Exception $e) {
+      throw new ConfigStorageException('Failed to write configuration to the database: ' . $this->$name, 0, $e);
+    }
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function delete($name) {
+    if (!$this->exists($name)) {
+      return FALSE;
+    }
+    db_delete($this->table, array('target' => $this->database))
+      ->condition('name', $name)
+      ->execute();
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function rename($name, $new_name) {
+    try {
+      db_delete($this->table, array('target' => $this->database))
+        ->condition('name', $new_name)
+        ->execute();
+      db_update($this->table, array('target' => $this->database))
+        ->fields(array('name' => $new_name))
+        ->condition('name', $name)
+        ->execute();
+    }
+    catch (\Exception $e) {
+      throw new ConfigStorageException('Failed to rename configuration from: ' . $name . ' to: ' . $new_name, 0, $e);
+    }
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getModifiedTime($name) {
+    $data = db_select($this->table, 'c', array('target' => $this->database))
+      ->fields('c', 'ctime')
+      ->condition('c.name', $name)
+      ->execute()
+      ->fetchField();
+
+    return empty($data) ? FALSE : $data;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function encode($data) {
+    $contents = backdrop_json_encode($data, TRUE);
+    if ($contents === FALSE) {
+      throw new ConfigStorageException(t('The configuration string could not be parsed.'));
+    }
+    return $contents;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function decode($raw) {
+    // Use json_decode() directly for efficiency.
+    $contents = json_decode($raw, TRUE);
+    if (is_null($contents)) {
+      throw new ConfigStorageException('The configuration string could not be parsed.');
+    }
+    return $contents;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function listAll($prefix = '') {
+    $results = db_select($this->table, 'c', array('target' => $this->database))
+      ->fields('c', array('name'))
+      ->condition('c.name', $prefix . '%', 'LIKE')
+      ->execute()
+      ->fetchAllAssoc('name', PDO::FETCH_ASSOC);
+    return array_column($results, 'name');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function deleteAll($prefix = '') {
+    $query = db_delete($this->table, array('target' => $this->database));
+    if ($prefix) {
+      $query->condition('name', $prefix . '%', 'LIKE');
+    }
+    $query->execute();
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function importArchive($file_uri) {
+    $realpath = backdrop_realpath($file_uri);
+
+    try {
+      $archiver = new ArchiverTar($realpath);
+      $archiver->extract($this->directory, $archiver->listContents());
```

Comment:
`$this->directory` does not exist. I'm betting that importing an archive doesn't work yet. It would need to first extract the archive to a temporary directory (see `exportArchive()` on creating that, and then it might need to truncate and insert the config? Or is truncating a problem? If so then it needs something more complex - comparing the archive to live and updating/deleting/inserting depending on the situation.

Reply:
Yes, good catch. I'll have to think this one through a little more.